Bio/SeqAnalysis/SeqAnalyzer.py

```diff
@@ -27,26 +30,31 @@ class SeqAnalyzer:
     def __init__(self, data: dict):
         """
         This method is called when an object is created from the class and it allows the class to initialize and
-        calculate via staticmethods usage(median_standard_deviation_average_length,  main_calculations
-        terminal_aa_counter) attributes of a SeqAnalyzer
+        calculate via static methods usage(median_standard_deviation_average_length, main_calculations
+        terminal_aa_counter) attributes of a SeqAnalyzer.
 
         :param data: takes in a dict with protein IDs as keys and SeqRecord objects as values
         """
-        id_seq_dictionary = {key: value.seq[0] for key, value in data}
+
+        id_seq_dictionary = {
+            key: value for key, value in {key: str(value.seq) for key, value in data.items()}.items() if
+            not (set('BZXJ').intersection(value)) and len(value) >= 10
+                            }
```

Comment:
Try to not use generic names `key`, `value` etc. I think you may want to reformat these lines to something more readable like this:
```
sequences = {seq_id: str(seq_data.seq) for seq_id, seq_data in data.items()}
id_seq_dictionary = {
    seq_id: seq
    for seq_id, seq in sequences.items()
    if not (set('BZXJ').intersection(seq)) and len(seq) >= 10
}
```